### PR TITLE
Core/BattlePets: Improvements when enabling battle pets

### PR DESCRIPTION
**Pull request by meji46**
**Changes proposed:**

- Defined BattlePetSlot enum class
- Enabled the battle pet system for new characters if it's enabled on the bnet account (slot 0 unlocked)
- Learn spells associated with the battle pet system

**Issues addressed:**
None

**Tests performed:**
Builds, tested in-game.

**Known issues and TODO list:** (add/remove lines as needed)
None

### CODE REVIEW

File: src/server/game/BattlePets/BattlePetMgr.h [hunk 1]
```diff
@@ -69,6 +74,15 @@ enum class BattlePetError : uint8
     TooHighLevelToUncage   = 7  // This pet is too high level for you to uncage.
 };
 
+enum class BattlePetSlot : uint8
+{
+    Slot0 = 0,
+    Slot1 = 1,
+    Slot2 = 2,
+
+    Count
```
Comment (MaxtorCoder) [inline src/server/game/BattlePets/BattlePetMgr.h:83]:
```suggestion
    Max
```
^ would be a better name I guess?

Comment (meji46) [reply src/server/game/BattlePets/BattlePetMgr.h:83]: I've used the same name as the enum classes in DBCEnums.h (https://github.com/TrinityCore/TrinityCore/blob/master/src/server/game/DataStores/DBCEnums.h#L292). I don't know if there are any criteria for choosing this name

Comment (MaxtorCoder) [reply src/server/game/BattlePets/BattlePetMgr.h:83]: Well imo it's the last entry, which is also `Max`. Let's see what the others say.

**Comment by Shauren** [comment]
Just curious, did you have a sniff of unlocking the pet battles?

File: sql/updates/world/master/9999_99_99_99_world.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+DELETE FROM `spell_learn_spell` WHERE `entry` = 125610;
+INSERT INTO `spell_learn_spell` (`entry`, `SpellID`, `Active`) VALUES
+(125610,119467,1),
+(125610,122026,1),
+(125610,125439,1);
```
Comment (Aokromes) [inline sql/updates/world/master/9999_99_99_99_world.sql:5]: missing newline.

**Review by Aokromes — APPROVED** [review]

**Comment by acidmanifesto** [comment]
> Just curious, did you have a sniff of unlocking the pet battles?

I can see about sniffing on out if u want?

**Comment by meji46** [comment]
> I can see about sniffing on out if u want?

I've passed him the sniff before

**Comment by acidmanifesto** [comment]
> > I can see about sniffing on out if u want?
> 
> I've passed him the sniff before

Gotcha

**Pull request merged**